core/src/Kokkos_View.hpp

```diff
@@ -593,8 +590,8 @@ class View : public ViewTraits<DataType, Properties...> {
   template <typename V>
   friend struct Kokkos::Impl::ViewTracker;
 
-  view_tracker_type m_track;
   map_type m_map;
+  view_tracker_type m_track;
```

Comment:
Why did this swap places with `m_map`?

Reply:
The 2 parameter constructor for view tracker requires that the map be constructed first.

Reply:
This is a real nit, and I'm sorry for it, but could we leave a comment here above the field definition saying to avoid reordering the fields? I could totally see myself having some harebrained reason to want to swap these and getting deeply confused when changing a field order broke stuff

Reply:
It's not a nit. Any time field ordering is important, period, there should be a comment explaining why